File: webapi/pkg/app/interfaces/i_logger.go
package interfaces

import (
	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type LogField struct {
	Key   string
	Value interface{}
}

type ILogger interface {
	GetHandleFunc() gin.HandlerFunc
	Debug(msg string, fields ...zap.Field)
	Info(msg string, fields ...zap.Field)
	Warn(msg string, fields ...zap.Field)
	Error(msg string, fields ...zap.Field)
	Fatal(msg string, fields ...zap.Field)
}

File: inventory_ms/application/src/interfaces/i_logger.rs
pub trait ILogger {
    fn trace(&self, target: &str, msg: &str);
    fn debug(&self, target: &str, msg: &str);
    fn info(&self, target: &str, msg: &str);
    fn warn(&self, target: &str, msg: &str);
    fn error(&self, target: &str, msg: &str);
}
